firebase-installations/src/main/java/com/google/firebase/installations/remote/FirebaseInstallationServiceClient.java

```diff
@@ -161,6 +164,8 @@ public InstallationResponse createFirebaseInstallation(
           return readCreateResponse(httpURLConnection);
         }
 
+        Log.w(FIS_TAG, readErrorResponse(httpURLConnection));
+
         if (httpResponseCode == 429 || (httpResponseCode >= 500 && httpResponseCode < 600)) {
           retryCount++;
           continue;
```

Comment:
I would ask you to please add another log message to the case that the request fails as `ResponseCode.BAD_CONFIG`, when we know that the application is misconfigured.
Since this is the main concern we are addressing, I recommend to add another log message, preferably as "Error" or as "Critical" - I'm not sure which one is more accurate:

```
Log.e/c(FIS_TAG, "Firebase Installations can not communicate with Firebase server APIs due to invalid configuration. Please update your Firebase initialization process and set valid Firebase options (API key, Project ID, Application ID) when initializing Firebase."
```

Reply:
Done. PTAL